addons/web/static/src/search/group_by_menu/group_by_menu.js

```diff
@@ -6,15 +6,14 @@ import { CustomGroupByItem } from "./custom_group_by_item";
 import { FACET_ICONS, GROUPABLE_TYPES } from "../utils/misc";
 import { sortBy } from "@web/core/utils/arrays";
 import { useBus } from "@web/core/utils/hooks";
+import { filterPropsForComponent } from "@web/core/utils/props";
 
 const { Component } = owl;
 
 export class GroupByMenu extends Component {
     setup() {
         this.icon = FACET_ICONS.groupBy;
-        this.dropdownProps = Object.keys(this.props)
-            .filter((key) => key in Dropdown.props)
-            .reduce((obj, key) => ({ ...obj, [key]: this.props[key] }), {});
+        this.dropdownProps = filterPropsForComponent(Dropdown, this.props);
```

Comment:
I'm not sure about this. It's a bit tedious, but what about filtering out props that are specific to the groupBy menu (in this case it's simple, there's only `showCaretDown`)? I'm not comfortable with using that kind of function (I would probably not define an utils to do that actually), because you kind of partially skip some parts of the validation with this (for instance, giving a wrong props could not be detected, because it would be filtered out).